banzai_nres/rv.py

```diff
@@ -0,0 +1,133 @@
+from banzai.stages import Stage
+from banzai.frames import ObservationFrame
+from banzai import dbs
+import pkg_resources
+from astropy.io import fits
+import numpy as np
+from astropy.table import Table
+from astropy import constants
+from astropy.time import Time
+from astropy.coordinates import SkyCoord, EarthLocation, solar_system_ephemeris
+from astropy import units
+from banzai.utils import stats
+import logging
+
+
+logger = logging.getLogger('banzai')
+
+# Speed of light in km/s
+c = constants.c.to('km / s').value
+
+
+def cross_correlate(velocities, wavelength, flux, flux_uncertainty, template_wavelength, template_flux):
+    """
+    :param velocities: in km / s
+    :param wavelength:
+    :param flux:
+    :param flux_uncertainty:
+    :param template_wavelength:
+    :param template_flux:
+    :return:
+
+    This is consistent with what we did in the satellite pipeline and
+    Zackay & Ofek, 2017, ApJ, 836, 187
+    """
+    x_cor = []
+    # for steps in 1 km/s from -2000 to +2000 km/s
+    # calculate the variance ahead of time
+    variance = flux_uncertainty * flux_uncertainty
+
+    for v in velocities:
+        doppler = 1.0 / (1.0 + v / c)
+        kernel = np.interp(doppler * wavelength, template_wavelength, template_flux)
+        correlation = kernel * flux
+        correlation /= variance
```

Comment:
posting a lingering reminder that we should compare RV scatter with and without variance division.

Reply:
But probably in a later PR.